### PR TITLE
Feat: add annotations and labels to gateway.cue

### PR DESCRIPTION
**Pull request by YanRyan**
add annotations and labels to gateway.cue and update gateway.yaml by "make reviewable"
fix this issue https://github.com/kubevela/kubevela/issues/6396

### CODE REVIEW

**Review by StevenLeiZhang — APPROVED** [review]
LGTM

**Comment by mouuii** [comment]
LGTM

File: vela-templates/definitions/internal/trait/gateway.cue [hunk 1]
```diff
@@ -168,5 +176,11 @@ template: {
 
 		// +usage=Specify a pathType for the ingress rules, defaults to "ImplementationSpecific"
 		pathType: *"ImplementationSpecific" | "Prefix" | "Exact"
+
+		// +usage=Specify the annotations to be added to the ingress
+		annotations?: [string]: string | null
```
Comment (FogDong) [inline vela-templates/definitions/internal/trait/gateway.cue:181]:
```suggestion
		annotations?: [string]: string
```

File: vela-templates/definitions/internal/trait/gateway.cue [hunk 2]
```diff
@@ -168,5 +176,11 @@ template: {
...
+
+		// +usage=Specify the labels to be added to the ingress
+		labels?: [string]: string | null
```
Comment (FogDong) [inline vela-templates/definitions/internal/trait/gateway.cue:184]:
```suggestion
		labels?: [string]: string
```

File: vela-templates/definitions/internal/trait/gateway.cue [hunk 3]
```diff
@@ -104,6 +104,14 @@ template: {
 				if parameter.gatewayHost != _|_ {
 					"ingress.controller/host": parameter.gatewayHost
 				}
+				for key, value in parameter.annotations {
+					"\(key)": "\(value)"
+				}
+			}
+			labels: {
+				for key, value in parameter.labels {
```
Comment (FogDong) [inline vela-templates/definitions/internal/trait/gateway.cue:112]:
```suggestion
				if parameter.labels !=  _|_ for key, value in parameter.labels {
```

File: vela-templates/definitions/internal/trait/gateway.cue [hunk 4]
```diff
@@ -104,6 +104,14 @@ template: {
 				if parameter.gatewayHost != _|_ {
 					"ingress.controller/host": parameter.gatewayHost
 				}
+				for key, value in parameter.annotations {
```
Comment (FogDong) [inline vela-templates/definitions/internal/trait/gateway.cue:107]:
```suggestion
				if parameter.annotations != _|_ for key, value in parameter.annotations {
```

File: vela-templates/definitions/internal/trait/gateway.cue [hunk 5]
```diff
@@ -104,6 +104,14 @@ template: {
...
+					"\(key)": "\(value)"
+				}
+			}
+			labels: {
```
Comment (FogDong) [inline vela-templates/definitions/internal/trait/gateway.cue:111]: Use `if parameter.labels !=  _|_ ` to judge here first.

**Review by FogDong — APPROVED** [review]

**Pull request merged**